lib/compilers/rust.js

```diff
@@ -27,9 +27,43 @@ import path from 'path';
 import _ from 'underscore';
 
 import { BaseCompiler } from '../base-compiler';
+import { eachLine } from '../utils';
 
 import { RustParser } from './argument-parsers';
 
+function parseOutput(lines, inputFilename, pathPrefix) {
+    const re = /^ --> <source>[(:](\d+)(:?,?(\d+):?)?[):]*\s*(.*)/;
+    const result = [];
+    eachLine(lines, function (line) {
+        line = line.split('<stdin>').join('<source>');
+        if (pathPrefix) line = line.replace(pathPrefix, '');
+        if (inputFilename) {
+            line = line.split(inputFilename).join('<source>');
+
+            if (inputFilename.indexOf('./') === 0) {
+                line = line.split('/home/ubuntu/' + inputFilename.substr(2)).join('<source>');
+                line = line.split('/home/ce/' + inputFilename.substr(2)).join('<source>');
+            }
+        }
+        if (line !== null) {
+            const lineObj = {text: line};
+            const asciiRe = /\x1B\[[\d;]*[Km]/g;
```

Comment:
Pretty cute `re` to remove ANSI colours, right? Maybe rename, as `asciiRe` implies something different to me; maybe `dropAnsiRe`?